### PR TITLE
Add 2D cape rendering

### LINKED ISSUE #22: [Idea] Have a cropped cape render
I think this would be really cool and I haven’t seen this anywhere else.
EE4BADC5-644A-4BE2-83F1-81F938DF50F7

### PR DESCRIPTION
**Pull request by TibiNonEst**
Closes #22 

- Adds cape render type & request kind
- Adds TextureKind enum, holds values SKIN or CAPE
- Refactors service.ts to support passing around both skin and cape texture data
- Removes "skin" from function names that produce both skin and cape textures in service.ts

### CODE REVIEW

**Comment by TibiNonEst** [comment]
There are 2 main things that still need to be considered with this pr:

1. Not all users have cape textures. How are we expressing failures to find cape textures for a user? For skins, we just pass back a default steve texture, but we probably don't want a default cape texture. The current implementation passes back JSON data saying the cape couldn't be found but does not change the Content-Type header from 'image/png', meaning browsers say it's a broken image. I could go refactor it more, moving around where headers are added to responses, but that seems counterintuitive until we decide how error responses will happen.

2. Do we want to include some kind of *raw cape* route. This could be useful for people who want the cape texture without a cape render, but I'm unsure about what such a route would be, or whether we want one in the first place. Something like `/cape/raw` could be good, but I'm not that comfortable doing multi-level routes with the current routing system, even doing `/armor/` for overlays was a stretch.

**Comment by astei** [comment]
> Not all users have cape textures. How are we expressing failures to find cape textures for a user?

I think returning a 404 makes sense.

> Do we want to include some kind of raw cape route?

I'm not sure if it makes sense. It wouldn't be hard to bolt such an endpoint on if someone requests it.

> I'm not that comfortable doing multi-level routes with the current routing system

Indeed, the current routing system is very simplistic. I'll have to queue a redo of it.

File: worker/index.ts [hunk 1]
```diff
@@ -100,6 +105,18 @@ async function processRequest(skinService: MojangRequestService, interpreted: Cr
         case RequestedKind.Skin: {
             return await skinService.retrieveSkin(interpreted, gatherer);
         }
+        case RequestedKind.Cape: {
+            const cape = await skinService.retrieveCape(interpreted, gatherer);
+            if (cape.status === 404) {
+                return new Response(JSON.stringify({ error: "User cape could not be found"}), {
```
Comment (astei) [inline worker/index.ts:111]: Now that I know this better, maybe it's better to return a 404 with a blank 1x1 image here. Would be interesting to see if the browser could handle this gracefully.

Comment (TibiNonEst) [reply worker/index.ts]: Pushed, seems browsers have no problem rendering a single transparent pixel in my testing.

**Pull request merged**